src/seed_clusterer.cpp

```diff
@@ -916,53 +943,54 @@ for (cluster_t c : chain_clusters) {
                         for (size_t j = 0 ; j <= i ; j++) {
                             //Combine clusters
                             if (old_cluster_indices[j].size() > 0) {
-                            combined_cluster[j].insert(
-                                old_cluster_indices[j].begin(),
-                                old_cluster_indices[j].end());
+                                new_cluster_indices.front().first[j].insert(
+                                      make_move_iterator(old_cluster_indices[j].begin()),
+                                      make_move_iterator(old_cluster_indices[j].end()));
                             }
                         }
                     } else {
-
+                        combined_clusters_indices[k].first.emplace_back();
                         new_cluster_indices.push_back(
-                                                  combined_clusters_indices[k]);
+                                                  move(combined_clusters_indices[k]));
                     } 
                     
                 }
-                combined_cluster[i].insert(c);
-                new_cluster_indices.push_back(make_pair(combined_cluster, 
-                                                         new_dists));
+                new_cluster_indices.front().first[i].insert(c);
+                new_cluster_indices.front().first.emplace_back();
+                new_cluster_indices.front().second = new_dists;
                 combined_clusters_indices = move(new_cluster_indices);
                 
             }
         }
         
         //Create actual clusters from indices
```

Comment:
Keep clusters that extend farthest left and right at the top of the vector of clusters